xwiki-platform-core/xwiki-platform-rest/xwiki-platform-rest-server/src/main/java/org/xwiki/rest/internal/resources/classes/PageClassPropertyValuesProvider.java

```diff
@@ -95,19 +95,26 @@ protected PropertyValues getAllowedValues(PageClass pageClass, int limit, String
 
     private String getLabel(EntityReference entityReference)
     {
-        String label;
+        String label = null;
         try {
             XWikiContext xcontext = this.xcontextProvider.get();
-            XWikiDocument document;
-            document = xcontext.getWiki().getDocument(entityReference, xcontext).getTranslatedDocument(xcontext);
-            label = document.getRenderedTitle(Syntax.PLAIN_1_0, xcontext);
+            XWikiDocument document = xcontext.getWiki().getDocument(entityReference, xcontext);
+            if (document != null) {
```

Comment:
In the discussion we had I tough you were using the public `com.xpn.xwiki.api.XWiki` API like most REST services do but with `com.xpn.xwiki.XWiki#getDocument` it's indeed impossible to end up with null since there is no right check. Either you get a XWikiDocument or it throw an exception.